weave/trace/op.py

```diff
@@ -407,16 +414,27 @@ def _do_call(
     if not pargs:
         pargs = _default_on_input_handler(op, args, kwargs)
 
-    if settings.should_disable_weave():
-        res = func(*pargs.args, **pargs.kwargs)
-    elif weave_client_context.get_weave_client() is None:
-        res = func(*pargs.args, **pargs.kwargs)
-    elif not op._tracing_enabled:
+    skip_tracing = (
+        settings.should_disable_weave()
+        or weave_client_context.get_weave_client() is None
+        or not op._tracing_enabled
+        or not get_tracing_enabled()
+    )
+
+    if skip_tracing:
```

Comment:
I try to avoid this.  The former is easier to debug which branch you're in if you're unexpectedly skipping tracing